ethers-providers/src/transports/quorum.rs

```diff
@@ -160,23 +160,41 @@ impl<T> QuorumProviderBuilder<T> {
 }
 
 impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
-    /// Returns the block height that _all_ providers have surpassed.
-    ///
-    /// This is the minimum of all provider's block numbers
-    async fn get_minimum_block_number(&self) -> Result<U64, ProviderError> {
-        let mut numbers = join_all(self.providers.iter().map(|provider| async move {
+    /// Returns the block height that a _quorum_ of providers have reached.
+    async fn get_quorum_block_number(&self) -> Result<U64, QuorumError> {
+        let (oks, errs) = join_all(self.providers.iter().map(|provider| async move {
             let block = provider.inner.request("eth_blockNumber", QuorumParams::Zst).await?;
```

Comment:
If there's a single error here wouldn't this still cause `join_all` to cancel all other futures and return an error?

Reply:
I didn't read the details to verify but I find it doubtful because it returns a `Vec<Result<_>>` rather than a `Result<Vec<>>`